src/Engine/Component/Script.cpp

```diff
@@ -45,35 +45,41 @@ Json::Value Script::Save() const {
     return component;
 }
 
-void Script::FillPropertyMap() {
+void Script::AddToPropertyMap(std::string &name, int type, int size, void* data) {
 
-    int propertyCount = instance->GetPropertyCount();
+    if(!IsInPropertyMap(name, type))
+        propertyMap[name] = new Property(type, size, data);
 
-    for (int n = 0; n < propertyCount; n++) {
+}
+
+void Script::CopyDataFromPropertyMap(const std::string &name, void* target){
 
-        int typeId = instance->GetPropertyTypeId(n);
-        void *varPointer = instance->GetAddressOfProperty(n);
+    std::memcpy(target, propertyMap[name]->data, propertyMap[name]->size);
+    
+}
 
-        auto it = propertyMap.find(instance->GetPropertyName(n));
-        if (it != propertyMap.end())
-            if (propertyMap[instance->GetPropertyName(n)].first == typeId)
-                continue;
+void* Script::GetDataFromPropertyMap(const std::string &name){
 
-        int size = Managers().scriptManager->GetSizeOfASType(typeId, varPointer);
-        if (size != -1) {
+    return propertyMap[name]->data;
 
-            propertyMap[instance->GetPropertyName(n)] = std::pair<int, void*>(typeId, malloc(size));
-            memcpy(propertyMap[instance->GetPropertyName(n)].second, varPointer, size);
+}
+
+bool Script::IsInPropertyMap(const std::string &name, int type) {
+
+    auto it = propertyMap.find(name);
+
+    if (it != propertyMap.end() && it->second->typeID == type)
+        return true;
+
+    return false;
 
-        }
-    }
 }
 
 /// Clears the property map.
 void Script::ClearPropertyMap() {
-
-    for (auto pair : propertyMap)
-        free(pair.second.second);
+    
+    for(auto pair : propertyMap) 
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'for' not followed by a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)